modifications/DataClass.h

```diff
@@ -13,85 +13,143 @@
 #include <qregexp.h>
 #include "GlobalDictionary.h"
 #include "QDir"
+#include <QDateTime>
+#include <SignatureVerifier.h>
+#include <QMessageBox>
 
-///This is the core class that handles the communications with the management server, stores all the data related to vehicle.
-/// This class is a singleton and never exposed to UI
-/// The only instance of this class is made in "QGCApplication.h" file.
 class DataClass : public QObject
 {
     Q_OBJECT
 public:
     explicit DataClass(QObject *parent = nullptr);
     QString getURL(){return serverUrl;}
+    QStringList getAllPlans(){
+        QStringList allPlans = {};
+        for(int i=0;i<flightData.length();i++){
+            allPlans.append(flightData[i].planName);
+        }
+        return allPlans;
+    }
+    QJsonObject getSelectedPlan(int selectedPlanIndex){
+        if(selectedPlanIndex<flightData.length()){
+            return flightData[selectedPlanIndex].plan;
+        }
+        return QJsonObject();
+    }
     bool checkdroneIDChanged(QString vehicleID);
 
+    void showMessage(QString msg, int errorCode);
+    QString errorCodeToString(int errorCode){
+        switch (errorCode) {
+            case -1: return "";
+            case 0: return "\nError: Please check your internet connection";
+            case 400: return "\nError: Bad Request";
+            case 401: return "\nError: Unauthorized Request";
+            case 404: return "\nError: Not Found";
+            case 500: return "\nError: Internal Server Error";
+            case 502: return "\nError: Bad Gateway";
+            case 503: return "\nError: Service Unavailable";
+            default: return  QString("Unknown Error Occured\nHttpResponse Code:") + QString(errorCode);
+        }
+    }
+
 signals:
-    ///This signal is emitted when authentication token is generated for further communications with management server
     void tokenGenerated();
-    ///This signal is emitted when authentication token is not generated for further communications with management server
```

Comment:
Again, why remove all these? Is there a different way to do documentation planned which would conflict?